### PR TITLE
docs(migrations): Starter Info for Migration Modes

### PR DESCRIPTION
**Pull request by ayirr7**
This doc has some information on how to get started with, and how to switch between the two migration modes (local and distributed).

### CODE REVIEW

File: docs/source/migrations/modes.rst [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+In your local Sentry environment configuration, set ``SENTRY_DISTRIBUTED_CLICKHOUSE_TABLES``
+to False. Start up the corresponding Clickhouse container (``sentry devservices up clickhouse``).
+
+Ensuring that your local configuration is prepared for migrations (reverting any existing
+migrations, or dropping certain databases/tables), run migrations as expected
+(``snuba migrations migrate --force``).
+
+
+Enabling Distributed Mode
```
Comment (lynnagara) [inline docs/source/migrations/modes.rst:29]: Could you add a note that this is experimental and isn't yet supposed to work properly? Since there are quite a number of known bugs, we should prevent anyone from attempting to use this for anything other than dev yet.

Comment (ayirr7) [reply docs/source/migrations/modes.rst:29]: Will do, once the new switch is confirmed.

File: docs/source/migrations/modes.rst [hunk 2]
```diff
@@ -0,0 +1,46 @@
+======================
+Snuba Migration Modes
+======================
+
+The main "switch" between the two modes for running data migrations (local and
+distributed) lives in a user's local Sentry configuration (in their ``sentry.conf.py``).
+The controlling envrionment variable is ``SENTRY_DISTRIBUTED_CLICKHOUSE_TABLES``,
+and its default value is set `at this line <https://github.com/getsentry/sentry/blob/master/src/sentry/conf/server.py#L127>`_.
```
Comment (fpacifici) [inline docs/source/migrations/modes.rst:8]: I would not add a link to the server.py file if not strictly necessary. 
That file change, the link would not so it will eventually point to the wrong place.
Giving the name of the setting is enough here.

Comment (ayirr7) [reply docs/source/migrations/modes.rst]: Good point; I removed the link.

File: docs/source/migrations/modes.rst [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+
+Enabling Distributed Mode
+============================
+
+In your local ``server.py``, set ``SENTRY_DISTRIBUTED_CLICKHOUSE_TABLES``
+to True. Start up the corresponding Clickhouse container (``sentry devservices up clickhouse``).
+
+Now, it is important to configure the clusters that can be used in Distributed tables. These are
+set in `this file <https://github.com/getsentry/sentry/blob/master/config/clickhouse/dist_config.xml>`_.
```
Comment (evanh) [inline docs/source/migrations/modes.rst:40]: The way this is worded suggests that I need to change something in this file before distributed mode will work. Is this true? Or is the default configuration already good enough?

Comment (ayirr7) [reply docs/source/migrations/modes.rst:40]: Thank you for raising this point - default configuration already works, and is probably best to work with at the moment. It is also possible, though, to write dist_config.xml such that there is more than 1 shard and replica, though that would also require understanding which storages can be sharded and which cannot be (will put out a PR to start some work on this soon). I'll edit the doc so these are noted.

**Review by evanh — COMMENTED** [review]
This looks good to me but I'll let Lyn give final approval.

File: docs/source/migrations/modes.rst [hunk 4]
```diff
@@ -0,0 +1,55 @@
+======================
+Snuba Migration Modes
+======================
+
+This doc outlines a basic way to try out distributed and local migrations.
+Note that this is experimental, and should be used only for development
+purposes at the moment.
```
Comment (lynnagara) [inline docs/source/migrations/modes.rst:7]: Can we rephrase this to say that only local (single node) migrations are currently supported. And it's only the distributed migrations that should only be used for development purposes. This line makes it sound like both modes are experimental.

File: docs/source/migrations/modes.rst [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+The main "switch" between the two modes for running data migrations (local and
```
Comment (lynnagara) [inline docs/source/migrations/modes.rst:9]:
```suggestion
If you are running ClickHouse via Sentry's devservices, the main "switch" between the two modes for running data migrations (local and
```
Though running ClickHouse via Sentry's devservices is common for Sentry devs, it's not necessarily the only way to run ClickHouse for Snuba.

File: docs/source/migrations/modes.rst [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+
+Enabling Local Mode
+=====================
+
+In your local ``server.py``, set ``SENTRY_DISTRIBUTED_CLICKHOUSE_TABLES``
+to False. Start up the corresponding Clickhouse container (``sentry devservices up clickhouse``).
+
+Ensuring that your local configuration is prepared for migrations (reverting any existing
+migrations, or dropping certain databases/tables), run migrations as expected
+(``snuba migrations migrate --force``).
```
Comment (lynnagara) [inline docs/source/migrations/modes.rst:30]: I don't think this is necessary. People should not manually revert migrations or drop tables/databases. They don't even need to run the migrate script manually since the devserver just does that.

File: docs/source/migrations/modes.rst [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+Finally, set up cluster connection details (for example, which storage is to be assigned
+to be which cluster) in `this file <https://github.com/getsentry/snuba/blob/master/snuba/settings/settings_distributed.py>`_.
+This needs to be done only for distributed migrations, as the default cluster details will be used in local mode.
+The default in this file works with the default cluster configurations mentioned above, so no changes
+are immediately necessary.
+
+Ensuring that your local configuration is prepared for migrations (reverting any existing
+migrations, or dropping certain databases/tables), run migrations with the ``SNUBA_SETTINGS``
+environment variable pointing to distributed mode. This can be done as follows:
+``SNUBA_SETTINGS=distributed snuba migrations migrate --force``.
```
Comment (lynnagara) [inline docs/source/migrations/modes.rst:55]: Similarly to above, no need to manually revert or drop anything.

File: docs/source/migrations/modes.rst [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+container, alter the environment variable mentioned above, and then "turn on" the
+same container to be in the new mode.
+
+More information on migrations in general can be found `here <https://github.com/getsentry/snuba/blob/master/MIGRATIONS.md>`_.
+
+Enabling Local Mode
+=====================
+
+In your local ``server.py``, set ``SENTRY_DISTRIBUTED_CLICKHOUSE_TABLES``
+to False. Start up the corresponding Clickhouse container (``sentry devservices up clickhouse``).
```
Comment (lynnagara) [inline docs/source/migrations/modes.rst:26]: May be worth mentioning this is the default mode, and you shouldn't need to do do anything special to achieve this.

File: docs/source/migrations/modes.rst (hunk 4 above)
Comment (ayirr7) [reply docs/source/migrations/modes.rst:7]: Clarified it, thank you for pointing this out.

File: docs/source/migrations/modes.rst (hunk 5 above)
Comment (ayirr7) [reply docs/source/migrations/modes.rst:9]: Modified.

File: docs/source/migrations/modes.rst (hunk 8 above)
Comment (ayirr7) [reply docs/source/migrations/modes.rst:26]: Done.

File: docs/source/migrations/modes.rst (hunk 6 above)
Comment (ayirr7) [reply docs/source/migrations/modes.rst]: I tested it and you're right; removed it.

**Review by fpacifici — APPROVED** [review]
